purchase_order_block/models/purchase_order.py

```diff
@@ -0,0 +1,55 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 Eficent Business and IT Consulting Services S.L.
+# Copyright 2017 Serpent Consulting Services Pvt. Ltd.
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl.html).
+
+from odoo import api, fields, models, _
+from odoo.exceptions import ValidationError
+
+
+class PurchaseOrder(models.Model):
+    _inherit = "purchase.order"
+
+    po_block_id = fields.Many2one(
+        comodel_name='purchase.block.reason',
+        string='Purchase Block Reason')
+
+    released = fields.Boolean(copy=False, readonly=True)
+
+    empty_check = fields.Boolean(copy=False, readonly=True)
+
+    @api.model
+    def create(self, vals):
+        po = super(PurchaseOrder, self).create(vals)
+        if po.po_block_id:
+            po.message_post(body=_('Order \"%s\" blocked with reason'
+                                   ' \"%s\"') % (po.name,
+                                                 po.po_block_id.name))
+        else:
+            po.empty_check = True
+        return po
+
+    @api.multi
+    def button_release_block(self):
+        for order in self:
+            order.\
+                message_post(body=_('Order \"%s\" with block \"%s\" has now '
+                                    'released') % (order.name,
+                                                   order.po_block_id.name))
+            order.released = True
```

Comment:
Isn't it better to just remove the `po_block_id`? setting it to false. Any order without a `po_block_id` is released